inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/extract/TidbExtractNode.java

```diff
@@ -0,0 +1,212 @@
+     * @param properties The properties connect to tidb
+     * @param primaryKey The primark key connect to tidb
+     * @param tableName The table name connect to tidb
+     * @param url The url connect to tidb
+     * @param database The database connect to tidb
+     * @param username The username  connect to tidb
+     * @param password The password  connect to tidb
+     * @param bootstrapServers The bootstrapServers connect to kafka
+     * @param topic The topic connect to kafka
+     * @param codec The codec is topic data of format
+     * @param groupId The kafka topic groupId
+     */
+    @JsonCreator
+    public TidbExtractNode(@Nonnull @JsonProperty("id") String id,
+            @Nonnull @JsonProperty("name") String name,
+            @Nonnull @JsonProperty("fields") List<FieldInfo> fields,
+            @Nullable @JsonProperty("watermarkField") WatermarkField watermarkField,
+            @JsonProperty("properties") Map<String, String> properties,
+            @JsonProperty("primaryKey") String primaryKey,
+            @Nullable @JsonProperty("url") String url,
+            @Nonnull @JsonProperty("tableName") String tableName,
+            @Nonnull @JsonProperty("database") String database,
+            @Nonnull @JsonProperty("username") String username,
+            @Nonnull @JsonProperty("password") String password,
+            @Nonnull @JsonProperty("bootstrapServers") String bootstrapServers,
+            @Nonnull @JsonProperty("topic") String topic,
+            @Nonnull @JsonProperty("codec") String codec,
+            @JsonProperty("groupId") String groupId,
+            @JsonProperty("autoOffsetReset") String autoOffsetReset
+    ) {
+        super(id, name, fields, watermarkField, properties);
+        this.url = Preconditions.checkNotNull(url, "url is null");
+        this.tableName = Preconditions.checkNotNull(tableName, "tableName is null");
+        this.database = Preconditions.checkNotNull(database, "database is null");
+        this.username = Preconditions.checkNotNull(username, "username is null");
+        this.password = Preconditions.checkNotNull(password, "password is null");
+        this.bootstrapServers = Preconditions.checkNotNull(bootstrapServers, "bootstrapServers is null");
+        this.topic = Preconditions.checkNotNull(topic, "topic is null");
+        this.codec = Preconditions.checkNotNull(codec, "codec is null");
+        this.groupId = groupId;
+        this.primaryKey = primaryKey;
+        this.autoOffsetReset = autoOffsetReset;
+    }
+
+    @Override
+    public String genTableName() {
+        return String.format("table_%s", super.getId());
+    }
+
+    @Override
+    public String getPrimaryKey() {
+        return primaryKey;
+    }
+
+    @Override
+    public Map<String, String> tableOptions() {
+        Map<String, String> options = super.tableOptions();
+
+        options.put("connector", "tidb");
```

Comment:
Maybe it is better to extract constants for the constant string?

Reply:
ok i will change it